client/src/modules/vouchers/modals/search.modal.js

```diff
@@ -13,59 +14,76 @@ VoucherRegistrySearchModalController.$inject = [
  * returning it as a JSON object to the parent controller.  The data can be
  * preset by passing in a filters object using filtersProvider().
  */
-function VoucherRegistrySearchModalController(ModalInstance, filters, Notify, moment, bhConstants) {
+function VoucherRegistrySearchModalController(ModalInstance, filters, Notify, moment, 
+bhConstants, Periods, Store, util, TransactionTypes, $translate) {
   var vm = this;
+  var changes = new Store({identifier : 'key'});
+  vm.filters = filters;
+  vm.searchQueries = {};
+  vm.defaultQueries = {};
 
-  // set controller data
-  vm.params = angular.copy(filters || {});
+  var searchQueryOptions = [
+    'reference', 'description', 'user_id',
```

Comment:
Adding `trans_id` here will allow the transaction ID to be loaded when the user opens the search modal for a second time.